src/app/InteractionModelEngine.h

```diff
@@ -133,19 +155,21 @@ class InteractionModelEngine : public Messaging::ExchangeDelegate
 
     InteractionModelEngine(void);
 
-    CHIP_ERROR Init(Messaging::ExchangeManager * apExchangeMgr);
+    CHIP_ERROR Init(Messaging::ExchangeManager * apExchangeMgr, void * const apAppState, const EventCallback aEventCallback);
```

Comment:
Could you add a comment about apAppState and what its lifetime is?

I have a hard time following the void * states and am wondering if we could be more typesafe. Generally a pair of 'event callback + void * appstate' can be type-safe represented with a delegate. Could that be done here?

Reply:
Agree, done, thanks